src/typed-geometry/functions/objects/intersection.hh

```diff
@@ -1256,10 +1288,36 @@ template <int D, class ScalarT>
 template <int D, class ScalarT>
 [[nodiscard]] constexpr optional<segment<D, ScalarT>> intersection(segment<D, ScalarT> const& a, sphere<D, ScalarT> const& b)
 {
-    static_assert(always_false<ScalarT>, "not implemented");
-    (void)a;
-    (void)b;
-    return {}; // TODO
+    // early-out: both segment points inside the sphere
+    if ((distance(a.pos0, b.center) < b.radius) && (distance(a.pos1, b.center) < b.radius))
```

Comment:
distance_sqr and radius_sqr are probably faster, right?

Reply:
yes :)